examples/4_train_policy_with_script.md

```diff
@@ -0,0 +1,157 @@
+This tutorial will explain the training script, how to use it, and particularly the use of Hydra to configure everything needed for the training run.
+
+## The training script
+
+LeRobot offers a training script at [`lerobot/scripts/train.py`](../../lerobot/scripts/train.py). At a high level it does the following:
+
+- Loads a Hydra configuration file for the following steps (more on Hydra in a moment).
+- Makes a simulation environment.
+- Makes a dataset corresponding to that simulation environment.
+- Makes a policy.
+- Runs a standard training loop with forward pass, backward pass, optimization step, and occasional logging, evaluation (of the policy on the environment), and checkpointing.
+
+## Our use of Hydra
+
+Explaining the ins and outs of [Hydra](https://hydra.cc/docs/intro/) is beyond the scope of this document, but here we'll share the main points you need to know.
+
+First, consider that `lerobot/configs` might have a directory structure like this (this is the case at the time of writing):
+
+```
+.
+├── default.yaml
+├── env
+│   ├── aloha.yaml
+│   ├── pusht.yaml
+│   └── xarm.yaml
+└── policy
+    ├── act.yaml
+    ├── diffusion.yaml
+    └── tdmpc.yaml
+```
+
+**_For brevity, in the rest of this document we'll drop the leading `lerobot/configs` path. So `default.yaml` really refers to `lerobot/configs/default.yaml`._**
+
+When you run the training script, Hydra takes over via the `@hydra.main` decorator. If you take a look at the `@hydra.main`'s arguments you will see `config_path="../configs", config_name="default"`. This means Hydra looks for `default.yaml` in `../configs` (which resolves to `lerobot/configs`).
+
+Among regular configuration hyperparameters like `device: cuda`, `default.yaml` has a `defaults` section. It might look like this.
+
+```yaml
+defaults:
+  - _self_
+  - env: pusht
+  - policy: diffusion
+```
+
+So, Hydra will grab `env/pusht.yaml` and `policy/diffusion.yaml` and incorporate their configuration parameters (any configuration parameters already present in `default.yaml` are overriden).
+
+## Running the training script with our provided configurations
+
+If you want to train Diffusion Policy with PushT, you really only need to run:
+
+```bash
+python lerobot/scripts/train.py
+```
+
+That's because `default.yaml` already defaults to using Diffusion Policy and PushT. To be more explicit, you could also do the following (which would have the same effect):
+
+```bash
+python lerobot/scripts/train.py policy=diffusion env=pusht
+```
+
+If you want to train ACT with Aloha, you can do:
```

Comment:
You can also mention the running time.

```suggestion
To reproduce ACT results on Aloha (TODO say the environement and task), run this:
```

Reply:
As I've mentioned in other comments here, I want to keep the focus on "how do I use LeRobot". There are other resources for the specific details of reproducing SOTA models.

Reply:
I think there are only these use cases:
- reproduce SOTA results (+ tune some hyperparameters, which are a diff with respect to SOTA hyerparameters)
- train on your own dataset
- train your own model on Aloha dataset

I dont think people are interested about "train ACT with Aloha" per se.

Reply:
I added a comment at the start of the advanced tutorial to make it clear that it addresses your point 2. Point 1 is covered in this tutorial with the changes I've made. For point 3, this is really dependent on the model, but I think the advanced tutorial helps.